torchgeo/datasets/xview.py

```diff
@@ -282,3 +282,195 @@ def plot(
             plt.suptitle(suptitle)
 
         return fig
+
+
+class XView2DistShift(XView2):
+    """A subclass of the XView2 dataset designed to reformat the original train/test splits.
+
+    This class allows for the selection of particular disasters to be used as the
+    training set (in-domain) and test set (out-of-domain). The dataset can be split
+    according to the disaster names specified by the user, enabling the model to train
+    on one disaster type and evaluate on a different, out-of-domain disaster. The goal
+    is to test the generalization ability of models trained on one disaster to perform
+    on others.
+    """
+
+    binary_classes: ClassVar[tuple[str, str]] = ('background', 'building')
+
+    valid_disasters: ClassVar[list[str]] = [
+        'hurricane-harvey',
+        'socal-fire',
+        'hurricane-matthew',
+        'mexico-earthquake',
+        'guatemala-volcano',
+        'santa-rosa-wildfire',
+        'palu-tsunami',
+        'hurricane-florence',
+        'hurricane-michael',
+        'midwest-flooding',
+    ]
+
+    def __init__(
+        self,
+        root: str = 'data',
+        split: str = 'train',
```

Comment:
Not required, but I'm trying to convert more of our datasets to use proper type hints:
```suggestion
        split: Literal['train', 'test'] = 'train',
```
Where `Literal` is from `typing`. Not all str are valid, only these two.

Reply:
Liked it! Will use this